### ISSUE TITLE
ASP.NET 5: No service for type 'Microsoft.Data.Entity.Storage.IRelationalConnection' has been registered

### ISSUE THREAD

**Issue by emfluenceindia** [aspnet/dnx#3366]
I have created the following class. Method `GetOriginalParamters` is supposed to pull original parameter names from a SQL stored procedure:

```c#
public class MySqlHelper
    {
        private MyDataContext _ctx;

        public MySqlHelper()
        {
            //
        }
        public MySqlHelper(MyDataContext context)
        {
            _ctx = context;
        }

        /// <summary>
        /// We would pass serialized JS object without any intermediate class definition
        /// Help topic: http://procbits.com/2011/04/21/quick-json-serializationdeserialization-in-c
        /// </summary>
        /// <returns></returns>
        public List<DbParameter> CreateParameters(Dictionary<string, string> objSerialized)
        {
            List<DbParameter> paramCollection = new List<DbParameter>();

            foreach (KeyValuePair<string, string> item in objSerialized)
            {
                paramCollection.Add(new SqlParameter(item.Key, item.Value));
            }

            return paramCollection;
        }

        /// <summary>
        /// Gets original parameter names from a Stored Procedure. Return type List<string>.
        /// Purpose: To check whether Parameter name(s) passed from client (AJAX) match(es) with original paramters of the procedure being called
        /// </summary>
        /// <param name="procedureName"></param>
        /// <returns></returns>
        public List<string> GetOriginalParamters(string procedureName)
        {
            List<string> paramNames = new List<string>();
            using (var cmd = _ctx.Database.GetDbConnection().CreateCommand())
            {
                cmd.CommandText = procedureName;
                cmd.CommandType = CommandType.StoredProcedure;

                foreach (SqlParameter p in cmd.Parameters)
                {
                    if (paramNames.Contains(p.ParameterName)) continue;
                    paramNames.Add(p.ParameterName.ToLower());
                }
            }

            //Can we put this in Application Level Cache here for faster accessibility at later point?
            //Or should we serialize the same to some XML or JSON and when application starts we shall re-evaluate all procedures and re-create serialized object?
            return paramNames;
        }
    }
```

I am calling method `GetOriginalParamters` in the following way:
`ColemanSqlHelper helper = new ColemanSqlHelper(new Model.ColemanContext());
 List<string> paramNames = helper.GetOriginalParamters(ProcedureHelper.CartInsertCustomer);`

But getting the following error:

> ASP.NET 5: No service for type 'Microsoft.Data.Entity.Storage.IRelationalConnection' has been registered
![error-screen](https://cloud.githubusercontent.com/assets/11072678/13311260/e7c6908c-dbb2-11e5-8a40-48473626b080.png)

I have idea what to do next. Please help!

**Comment by Eilon** [comment]
@divega thoughts? Looks like something with EF?

**Comment by divega** [comment]
@emfluenceindia can you share your Startup.cs code? I think this could happen if EF and EF provider services are not wired up in `ConfigureServices()`.  

BTW,  most likely this issue is unrelated to DNX.

**Comment by emfluenceindia** [comment]
Hi divega,
Here is my code:
`public void ConfigureServices(IServiceCollection services)
{
services.AddEntityFramework()
                .AddSqlServer()
                .AddDbContext<MyDbContext>(options =>
                    options.UseSqlServer(Configuration["Data:DefaultConnection:ConnectionString"])
                );

            // Add framework services.
            services.AddApplicationInsightsTelemetry(Configuration);

            services.AddMvc();
            services.AddCaching();

            //CORS
            services.AddCors(options => options.AddPolicy("AllowAll",
                p => p.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader()));

            //Try to get stored proce parameter information here
            //MyDbContext context = new MyDbContext();

            /*
                Runtime error at Line # 71:
                An exception of type 'System.InvalidOperationException' occurred in Microsoft.Extensions.DependencyInjection.Abstractions.dll but was not handled in user code
                Additional information: No service for type 'Microsoft.Data.Entity.Storage.IRelationalConnection' has been registered.
            */

            //using (var cmd = _ctx.Database.GetDbConnection().CreateCommand())
            //{
            //    cmd.CommandText = StaticObjects.ProcedureHelper.CustomerAddressInsert;
            //    cmd.CommandType = System.Data.CommandType.StoredProcedure;

            //    foreach (SqlParameter p in cmd.Parameters)
            //    {
            //        Console.WriteLine(p.ParameterName);
            //        Console.WriteLine(p.DbType);
            //        Console.WriteLine(p.SqlDbType);
            //    }
            //}
}`

**Comment by divega** [comment]
@emfluenceindia The Startup code seems correct and it should allow having the dependency injection container create instances of `MyDbContext`. However I went back to your original post and I found that you are creating the DbContext outside the container:

``` C#
ColemanSqlHelper helper = new ColemanSqlHelper(new Model.ColemanContext()); List<string> paramNames = helper.GetOriginalParamters(ProcedureHelper.CartInsertCustomer); 
```

That won't work unless you have `ColemanContext` setup in a way that it can configure its own connection, e.g. by overriding the `OnConfiguring()` method as explained in http://docs.efproject.net/en/latest/platforms/full-dotnet/new-db.html. 

I will leave it up to you to decide whether you want to change your code, but my recommendation would be to resolve `ColemanContext` from DI.

I will file bugs for the exception message not being clear and for improving how EF interacts with DI.
